### DISCUSSION TITLE
How to calculate the binary cache filename without installing and building

### DISCUSSION THREAD

**Discussion by GordonSmith** [microsoft/vcpkg#29000] [Q&A]
Is there any way to pre-calculate the zip hash value prior to install / build?

**Comment by JackBoosY** [comment]
It's so difficult that I suggest you to install it with debug mode.

**Reply by GordonSmith** [reply]
How does vcpkg calculate it prior to download? If the logic is in vcpkg, could a new flag be added to expose it I wonder?

**Reply by JackBoosY** [reply]
You can use extra command argument `--debug` to show the related hashs.
Or you can just find them in _VCPKG_ROOT/buildtrees/PORT/TRIPLET.vcpkg_abi_info.txt_.

**Reply by Neumann-A** [reply]
> VCPKG_ROOT/buildtrees/PORT/TRIPLET.vcpkg_abi_info.txt.

requires a dependent port otherwise you don't get the full hash in that file as far as I am aware.
